packages/grid/x-data-grid/src/components/columnsManagement/GridColumnsManagement.tsx

```diff
@@ -0,0 +1,356 @@
+                  checked={allHideableColumnsVisible}
+                  indeterminate={!allHideableColumnsVisible && !allHideableColumnsHidden}
+                  onClick={() => toggleAllColumns(!allHideableColumnsVisible)}
+                  name={apiRef.current.getLocaleText('columnsManagementShowHideAllText')}
+                  size="small"
+                  {...rootProps.slotProps?.baseCheckbox}
+                />
+              }
+              sx={{
+                [`.MuiFormControlLabel-label`]: { textTransform: 'uppercase', fontSize: '14px' },
+              }}
+              label={apiRef.current.getLocaleText('columnsManagementShowHideAllText')}
+            />
+          ) : (
+            <span />
+          )}
+
+          {!disableResetButton ? (
+            <rootProps.slots.baseButton
+              onClick={() => apiRef.current.setColumnVisibilityModel(initialColumnVisibilityModel)}
+              disabled={isResetDisabled}
+              {...rootProps.slotProps?.baseButton}
+            >
+              Reset
+            </rootProps.slots.baseButton>
+          ) : null}
+        </GridColumnsManagementFooter>
+      ) : null}
+    </React.Fragment>
+  );
+}
+
+GridColumnsManagement.propTypes = {
+  // ----------------------------- Warning --------------------------------
+  // | These PropTypes are generated from the TypeScript type definitions |
+  // | To update them edit the TypeScript types and run "yarn proptypes"  |
+  // ----------------------------------------------------------------------
+  /**
+   * If `true`, the column search field will be focused automatically.
+   * If `false`, the first column switch input will be focused automatically.
+   * This helps to avoid input keyboard panel to popup automatically on touch devices.
+   * @default true
+   */
+  autoFocusSearchField: PropTypes.bool,
+  /**
+   * If `true`, the `Reset` button will not be disabled
+   * @default false
+   */
+  disableResetButton: PropTypes.bool,
+  /**
+   * If `true`, the `Show/Hide all` toggle checkbox will not be displayed.
+   * @default false
+   */
+  disableShowHideToggle: PropTypes.bool,
+  /**
+   * Returns the list of togglable columns.
+   * If used, only those columns will be displayed in the panel
+   * which are passed as the return value of the function.
+   * @param {GridColDef[]} columns The `ColDef` list of all columns.
+   * @returns {GridColDef['field'][]} The list of togglable columns' field names.
+   */
+  getTogglableColumns: PropTypes.func,
+  searchPredicate: PropTypes.func,
+  sort: PropTypes.oneOf(['asc', 'desc']),
+} as any;
+
+const GridColumnsManagementBody = styled('div', {
+  name: 'MuiDataGrid',
+  slot: 'ColumnsManagement',
+  overridesResolver: (props, styles) => styles.columnsManagement,
+})<{ ownerState: OwnerState }>(({ theme }) => ({
+  padding: theme.spacing(0, 3, 1.5),
+  display: 'flex',
+  flexDirection: 'column',
+  overflow: 'auto',
+  flex: '1 1',
+  maxHeight: 400,
```

Comment:
```suggestion
  maxHeight: 400,
  alignItems: 'flex-start',
```

To avoid this:

https://github.com/mui/mui-x/assets/13808724/cd9c672c-238b-4d5d-9605-6f23db7f96dd